### PR TITLE
Don't assume that the iframe is on the bottom-right

### PR DESCRIPTION
**Pull request by ngotchac**
Place the iframe and set the correct width and height of the Parity Signer Bar iframe on load and when the close event is fired.

This is the first step for paritytech/parity#4886
Second step is here : paritytech/parity#4987

### CODE REVIEW

File: web3/index.js [hunk 1]
```diff
@@ -35,24 +35,87 @@ isIntegrationEnabled()
...
-      opened: ev.detail.opened
-    }, '*');
+      opened
+    };
+
+    parityBarElement.style.maxHeight = '100vh';
+
+    // Resize the iframe if it's closing
+    if (!opened) {
+      return window.setTimeout(() => {
```
Comment (tomusdrw) [inline web3/index.js]: Would be better to use https://developer.mozilla.org/en-US/docs/Web/Events/transitionend

File: web3/index.js [hunk 2]
```diff
@@ -35,24 +35,87 @@ isIntegrationEnabled()
...
 
 /**
+ * Resize and position the iframe according
+ * to the Parity Bar style
+ */
+function resizeParityBar (wait = 2000) {
+  const parityBarElement = getParityBarElement();
+
+  // Try again in 100ms
+  if (!parityBarElement && wait > 0) {
```
Comment (tomusdrw) [inline web3/index.js]: Why is this needed?

File: web3/index.js [hunk 3]
```diff
@@ -35,24 +35,87 @@ isIntegrationEnabled()
...
+  }
+
+  const { height, width } = parityBarElement.getBoundingClientRect();
+  const { left, top } = parityBarElement.style;
+  const computedStyle = window.getComputedStyle(parityBarElement);
+  const iframeStyle = {};
+
+  if (left) {
+    iframeStyle.left = 0;
+    iframeStyle.width = (width + parseInt(computedStyle.left) + 2) + 'px';
```
Comment (tomusdrw) [inline web3/index.js]: `2` should be moved to constant (probably `BORDER_SIZE` or sth like this)

File: web3/content_script.js [hunk 4]
```diff
@@ -155,23 +155,41 @@ function main () {
...
+        case EV_SIGNER_BAR:
+          if (ev.data.opened) {
+            iframe.classList.add(styles.iframe__open);
+          } else {
+            iframe.classList.remove(styles.iframe__open);
+          }
+
+          break;
+
+        case EV_IFRAME_STYLE:
```
Comment (tomusdrw) [inline web3/content_script.js]: I think it would be better to give the event more specific name, since it's also closes the iframe and is fired after the resize happened (it's not simply updating the styles as the name suggests now).

File: web3/content_script.js [hunk 5]
```diff
@@ -155,23 +155,41 @@ function main () {
...
+
+      if (!ev.data.type) {
         return;
       }
-      if (ev.data.opened) {
-        iframe.classList.add(styles.iframe__open);
-      } else {
-        iframe.classList.remove(styles.iframe__open);
+
+      switch (ev.data.type) {
```
Comment (tomusdrw) [inline web3/content_script.js]: TBH I prefer a series of early-exiting ifs instead of `switch` (less identation, clearer flow, no risk of forgeting to break/return):
```js
if (type === EV_SIGNER_BAR) {
  // do something
  return;
}

if (type === EV_IFRAME_STYLE) {
  // do something
  return;
}
```
but I'm also ok with this approach if you give me one argument why you prefer it :)

File: web3/styles.css [hunk 6]
```diff
@@ -18,24 +18,22 @@
...
   /* Delay closing the iframe to perform background animation. */
   transition: width 0s linear .2s, height 0s linear .2s, background .2s ease-in;
 }
 
 .iframe__open {
-  height: 100vh;
-  width: 100vw;
+  bottom: 0 !important;
+  right: 0 !important;
+  height: 100vh !important;
```
Comment (tomusdrw) [inline web3/styles.css]: probably `bottom:0,top:0,left:0,right:0` would work as well, no?

**Review by tomusdrw — COMMENTED** [review]
Few grumbles.

File: web3/index.js (hunk 2 above)
Comment (ngotchac) [reply web3/index.js]: I added a longer comment. It is because on load, the parity bar is not available immediately. It takes time to show it, so we try every 100ms for 2 seconds to see if it's loaded.

File: web3/index.js (hunk 1 above)
Comment (ngotchac) [reply web3/index.js]: :+1:

**Comment by ngotchac** [comment]
Updated the PR ! Seems to work ok. To test : 

Run `NODE_ENV=production npm run build && node webpack/build.server.js` for the UI
Set `127.0.0.1:3000` in the Extension options

**Comment by tomusdrw** [comment]
Opening and Closing signer on older version of Parity embed code works very slowly.
(it opens fast, but after closing the modals stays open for couple of seconds).

**Comment by tomusdrw** [comment]
Issues I've found:
1. Slow closing of signer
2. Initial size is incorrect (too small) so the texts are not displayed (only icons). Actually this sometimes happens randomly, perhaps the size is not calculated correctly? What about some sane minimal values?
3. The bar flickers when dropped.

**Comment by tomusdrw** [comment]
@ngotchac How is the progress going on this?

**Comment by ngotchac** [comment]
1. Will look into this, we have to find a better way of closing, but it's not trivial since there could be some animations involved (thus closing slowly)
2. It has happened to me as well. It is a bit random, so some minimum would be good yes :)
3. I tried to mitigate this but couldn't find a way... :( But I think it was already happening before when opening the Signer, no ?

**Comment by tomusdrw** [comment]
1. Wouldn't using `ontransitionend` event solve that?
2. :+1:
3. Didn't notice that before, but possibly it was already happening before. We can log this as a separate issue to investigate.

**Comment by ngotchac** [comment]
Updated the PR. The flickering is annoying but couldn't find the reason...

**Comment by tomusdrw** [comment]
So the extensions stopped working correctly for me. I'm not sure if it was introduced by this PR (will do some more testing), but version from the store works correctly.

It complains that the transport is not initialized all the time and there are some errors regarding blocked resources.
